tokio/src/sync/mpsc/bounded.rs

```diff
@@ -255,6 +298,34 @@ impl<T> Sender<T> {
         }
     }
 
+    /// Blocking send to call outside of asynchronous contexts
+    pub fn blocking_send(&mut self, value: T) -> Result<(), SendError<T>> {
+        // TODO ensure can't be called from block_on
+        let mut park = CachedParkThread::new();
+        let waker = match park.get_unpark() {
+            Ok(s) => s.into_waker(),
+            Err(_) => return Err(SendError(value)),
+        };
+        let mut cx = Context::from_waker(&waker);
+
+        loop {
+            if let Poll::Ready(v) = self.poll_ready(&mut cx) {
```

Comment:
Seems like we should be able to just poll the async `send` function. That would deduplicate some code.

Reply:
So that thought crossed my mind, but value would be moved into `send` and then I can't use it when returning an error if `park.park()` fails. Unless there's a better way to get around that?

Reply:
It seems like we should just panic if `park.park()` fails. That would be a bug in our program.

Reply:
Yes, `park.park()` returning an error is legacy. Unwrap is fine. That said, I commented above that you can just use `enter.block_on`.